### ISSUE TITLE
[BUG] Initialisation fatally crashes app in v7.0.1

### ISSUE THREAD

**Issue by zigapovhe** [JaffaKetchup/flutter_map_tile_caching#93]
### What is the bug?

When I updated your package to v7.0.0 or v7.0.1 and migrated initialisation function, I get following errors when I try to compile my app: [Screenshot](https://i.imgur.com/ahnoiur.png)

It looks like method `whereNotNull` is not defined

### What is the expected behaviour?

To successfully compile my app. It worked before the package upgrade.

### How can we reproduce this issue?

/

### Do you have a potential solution?

/

### Can you provide any other information?

```
Flutter 3.7.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 9944297138 (3 days ago) • 2023-02-08 15:46:04 -0800
Engine • revision 248290d6d5
Tools • Dart 2.19.2 • DevTools 2.20.1

flutter_map_tile_caching: ^7.0.1
flutter_map: ^3.1.0
```

### Platforms Affected

Android, iOS

### Severity

Fatal: Causes the application to crash

### Frequency

Consistently: Always occurs at the same time and location

### Requirements

- [X] I agree to follow this project's [Code of Conduct](https://github.com/jaffaketchup/flutter_map_tile_caching/blob/main/CODE_OF_CONDUCT.md)
- [X] I have checked for known issues in the GitHub tracker and [on the docs site](https://fmtc.jaffaketchup.dev/known-issues-and-reporting#known-issues)
- [X] I have completed the pre-report checks listed [on the docs site](https://fmtc.jaffaketchup.dev/known-issues-and-reporting#reporting-a-bug)
- [X] If I am using FMTC in a proprietary project, I have read and understood [this important legal notice](https://fmtc.jaffaketchup.dev/#proprietary-licensing)

**Comment by JaffaKetchup** [comment]
Hey, thanks for the report :)
I'll investigate today and try to release a fix within the next 24 hrs. Please use v7 for now.

**Comment by JaffaKetchup** [comment]
Hi @zigapovhe,
I can't reproduce this, and GitHub Actions has successfully built the example application. It looks like their may be something off in your configuration?
Thanks for the report though, it has made me realise another small mistake that needs to be fixed.

**Comment by NomiAdam** [comment]
I had the exact same issue as stated by @zigapovhe. 

To fix the issue I had explicitly define `stream_transform: 2.1.0` in pubspect.yaml

Hope this might save others some digging

**Comment by JaffaKetchup** [comment]
@NomiAdam Thanks for joining in.

That's really strange, I've run the `flutter pub outdated` command, and GitHub Actions appropriately builds the project. I also know that not all users are having this issue, so there must be some kind of project config combination that causes this issue.
Are you doing anything unusual, like using Flutter's beta branch?

For now, I'll pin this to the top of the issue tracker, but there's nothing I can do about it.

**Comment by zigapovhe** [comment]
@JaffaKetchup Good news here. After @NomiAdam mentioned the `stream_transform` package, I've checked in my pubspec.lock file and noticed that the `stream_transform` package was at the v2.0.0 version. Updated your package to v7.1.0, run `flutter pub get`, no changes. Even if I'd run `flutter pub get `countless times, the `stream_transform` package was still at v2.0.0 (but other packages did change if there is an update available). Deleting pubspec.lock file and re-running `flutter pub get` fixed the issue without manually importing the stream_transform package. (After rechecking pubspec.lock file, I can confirm that the `stream_transform` package is now at v2.1.0) I can also confirm that all of your package v7 versions (including v7.0.0) work now as expected. Issue can be probably closed, since there is probably nothing wrong with your package but looks like some weird flutter bug?
